### ISSUE TITLE
Fonts, favicons and external libs management

### ISSUE THREAD

**Issue by nyxgear** [atuinframework/atuin-tools#4]
I would like to enhance the minification task in creating the `app/static/min` folder by making it aware of three others directories: `favicon`, `font` and `lib`.

This three folders and their contents would be copied as-is from the `src` folder to the `min` folder. 

```
static
├── min
│   ├── css
│   ├── favicon
│   ├── font
│   ├── img
│   ├── js
│   └── lib
└── src
    ├── css
    ├── favicon
    ├── font
    ├── img
    ├── js
    └── lib
```

**Advantages obtained**

- Fonts, favicons and external libraries support. Now not present.
- The three folders would be automatically copied and kept up to date into the `min` folder by the `watch` and `prepare-deploy` tasks.
- No more need to create ad-hoc solutions for projects using atuin-tools that need to import external libraries (css/js/img files), fonts or for which you want to set up a favicon.
- Carefree deployments without the worry the `min` folder has something lacking.

**Comment by xcash** [comment]
Agreed.

I would love to add also a `ts` folder alongside 'js'. But we need to define a strategy on what to generate as final result.
Should we concatenate it in the all.js after (or before) vanilla javascript?

**Comment by nyxgear** [comment]
I think this strictly depends on the strategy the `.ts` files code will adopt. 

Do we plan to create `.ts` files which will follow the pattern of functions activated according to the HTML class? If the answer is yes, and I think so, then it would be good to concatenate TypeScript files compilation outcome to the all.js file. After all, the produced JS code will be considered on the same level of the vanilla JS, right?

**Comment by xcash** [comment]
Yes. But not limited to that. As for vanilla, sometimes you create classes and utilities functions that are global. Library functions. 
And TS is great for complex code. 
We will migrate all the included atuin js code to TS. So *maybe TS should be the first*, as to provide everything needed by the "simpler" vanilla js code eventually used by the developer. 

This is really easy to do with TS as it will transpile seamlessy existing vanilla js.

**Comment by nyxgear** [comment]
Ok, looking at it from this perspective even to me seem that we should include the TS compiled output *before* the vanilla JS.

I would like to absolutely avoid to create another `.js` file to be included in the HTML before the actual `all.js`. I suggest to still keep the regular `all.js` file but to create it by the output of the compiled TS files then followed by the concatenated vanilla JS files.

**Comment by xcash** [comment]
Agreed :+1:
